app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/subscription/Sub02Controller.scala

```diff
@@ -112,12 +112,16 @@ class Sub02Controller @Inject() (
   // End of normal subscription journey
   def migrationEnd(service: Service): Action[AnyContent] = authAction.ggAuthorisedUserWithServiceAction {
     implicit request => user: LoggedInUserWithEnrolments =>
-      if (UserLocation.isRow(requestSessionData))
-        subscriptionDetailsService.cachedCustomsId flatMap {
-          case Some(_) => renderPageWithName(service)
-          case _       => renderPageWithNameRow(service)
-        }
-      else renderPageWithName(service)
+      activatedEnrolmentForService(user, service) match {
+        case Some(_) => Future.successful(Redirect(CompletedEnrolmentController.enrolSuccess(service)))
+        case _ =>
+          if (UserLocation.isRow(requestSessionData))
+            subscriptionDetailsService.cachedCustomsId flatMap {
+              case Some(_) => renderPageWithName(service)
+              case _       => renderPageWithNameRow(service)
+            }
+          else renderPageWithName(service)
```

Comment:
I'm thinking about this part to convert this to:
```
subscriptionDetailsService.cachedCustomsId flatMap {
    case None if UserLocation.isRow(requestSessionData) => renderPageWithNameRow(service)
    case _       => renderPageWithName(service)
}
```

but I'm not sure which is more readable.

Additionally with second approach we will always call mongo to retrieve the cachedCustomsId so probably better to keep it in the way it is